riscv/processor.cc

```diff
@@ -819,8 +833,13 @@ const char* processor_t::get_privilege_string()
 
 void processor_t::enter_debug_mode(uint8_t cause)
 {
+  const bool has_zicfilp = extension_enabled(EXT_ZICFILP);
   state.debug_mode = true;
-  state.dcsr->write_cause_and_prv(cause, state.prv, state.v);
+  state.dcsr->update_fields(cause, state.prv, state.v,
+                            (has_zicfilp ? state.elp : elp_t::NO_LP_EXPECTED));
+  if (has_zicfilp) {
+    state.elp = elp_t::NO_LP_EXPECTED;
```

Comment:
Why not just is if ( extension_enabled[EXT_ZICFILP] )?

Reply:
should not need has_zicfilp qualification here.

Reply:
> Why not just is if ( extension_enabled[EXT_ZICFILP] )?

Just felt that `extension_enabled(EXT_ZICFILP)` looks a bit expensive from the call site.

Fixed.